### PR TITLE
Time all the things

### PR DESCRIPTION
**Pull request by partouf**
API Request timing is displayed in the UI, but there have been some requests in the past to display more exact timings for the execution of user code (e.g. https://github.com/compiler-explorer/compiler-explorer/issues/2337)

Timing
- [x] Compiler
- [x] Executable
- [x] Library download
- [x] Binary from S3 cache

Display in UI
- [x] Compiler
- [x] Executable
- [x] Library download

### CODE REVIEW

File: test/exec-tests.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ function testExecOutput(x) {
     // Work around chai not being able to deepEquals with a function
     x.filenameTransform.should.be.a('function');
     delete x.filenameTransform;
+    delete x.execTime;
```
Comment (mattgodbolt) [inline test/exec-tests.js:32]: cunning :)

File: lib/buildenvsetup/ceconan.js [hunk 2]
```diff
@@ -91,8 +91,9 @@ export class BuildEnvSetupCeConanDirect extends BuildEnvSetupBase {
         });
     }
 
-    async downloadAndExtractPackage(downloadPath, packageUrl) {
+    async downloadAndExtractPackage(libId, version, downloadPath, packageUrl) {
```
Comment (mattgodbolt) [inline lib/buildenvsetup/ceconan.js:94]: do we use the libId and version? Or is that coming soon? (I just noticed this is WIP! sorry)

**Review by mattgodbolt — COMMENTED** [review]
(looks ok! just spotted it was WIP)

**Comment by partouf** [comment]
image

**Comment by partouf** [comment]
There are situations where the browser just returns the cached document and then the Network/JS time bar is negative. Not sure how to detect that situation reliably other than checking for time < 0 I guess.

**Comment by RubenRBS** [comment]
I love this

**Comment by partouf** [comment]
This is now live!

**Pull request merged**